api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java

```diff
@@ -173,4 +179,75 @@ public void deleteOrderGroupAttributeType_shouldDeleteOrderGroupAttributeTypeFro
 		dao.deleteOrderGroupAttributeType(orderGroupAttributeType);
 		assertNull(dao.getOrderGroupAttributeByUuid(UUID));
 	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getAllOrderAttributeTypes()}
+	 * @throws Exception
+	 */
+	@Test
+	public void getAllOrderAttributeTypes_shouldReturnAllOrderAttributeTypes() {
+		assertEquals(4, dao.getAllOrderAttributeTypes().size());
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderAttributeTypeByUuid(String)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderAttributeTypeByUuid_shouldReturnOrderAttributeTypeUsingProvidedUuid() {
+		assertEquals("Referral", dao.getOrderAttributeTypeByUuid("9758d106-79b0-4f45-8d8c-ae8b3f25d72a").getName());
+	}
+
+	@Test
+	public void saveOrderAttributeType_shouldSaveTheProvidedOrderAttributeTypeToDatabase() {
+		final Order order = Context.getOrderService().getOrder(1);
+		OrderAttributeType orderAttributeType = new OrderAttributeType();
```

Comment:
I think it is good to first show us here that the size of the rows or list was less by one e.g
`assertEquals(4,  dao.getAllOrderAttributeTypes().size());`
Then you can go ahead and save the new `OrderAttribute`

Reply:
Definitely ...I will address that.